foundry/foundry.py

```diff
@@ -504,8 +504,17 @@ def download(self, globus=True, verbose=False, **kwargs):
         """
         # Check if the dir already exists
         path = os.path.join(self.config.local_cache_dir, self.mdf["source_id"])
+
         if os.path.isdir(path):
```

Comment:
I would make it clear (either with a comment or with an else statement, if there are no unintended consequences) that if the dir is present but the incorrect number of files are in it, then the download would be re-attempted